mesonbuild/dependencies.py

```diff
@@ -1599,6 +1605,132 @@ def __init__(self, environment, kwargs):
     def get_link_args(self):
         return []
 
+class LLVMDependency(Dependency):
+    """LLVM dependency.
+
+    LLVM uses a special tool, llvm-config, which has arguments for getting
+    c args, cxx args, and ldargs as well as version.
+    """
+
+    # Ordered list of llvm-config binaries to try. Start with base, then try
+    # newest back to oldest (3.5 is abitrary), and finally the devel version.
+    llvm_config_bins = [
+        'llvm-config', 'llvm-config-4.0', 'llvm-config-3.9', 'llvm-config39',
+        'llvm-config-3.8', 'llvm-config38', 'llvm-config-3.7', 'llvm-config37',
+        'llvm-config-3.6', 'llvm-config36', 'llvm-config-3.5', 'llvm-config35',
+        'llvm-config-devel',
+    ]
+    llvmconfig = None
+    _llvmconfig_found = False
+    __best_found = None
+
+    def __init__(self, environment, kwargs):
+        super().__init__('llvm-config', kwargs)
+        # It's necessary for LLVM <= 3.8 to use the C++ linker. For 3.9 and 4.0
+        # the C linker works fine if only using the C API.
+        self.language = 'cpp'
```

Comment:
Does linking to llvm still pull in stdc++ with >=3.9 if you link with a C linker? If not, making this conditional on the version would make sense since it would avoid that runtime dependency.